Docs: make it clear that MariaDB is also supported as datasource

fauust opened the pull request:
**What is this feature?**

Documentation improvement to make it clear that MariaDB is a valid datasource.

**Why do we need this feature?**

MariaDB is the default relational DB on various important Linux distribution, it's important to make it clear that it's perfectly supported.

**Who is this feature for?**

MySQL/MariaDB datasource interested users.

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=71832) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=71832) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=71832) it.</sub>

fauust commented:
BTW I saw that there are integration tests in your drone CI for mysql5.7 and mysql8.0, I would be happy to add them for some MariaDB version if you think that this would make sense and complete this PR...

baldm0mma commented:
> BTW I saw that there are integration tests in your drone CI for mysql5.7 and mysql8.0, I would be happy to add them for some MariaDB versions if you think that this would make sense and complete this PR...

I think those would be great to add!

fauust commented:
> > BTW I saw that there are integration tests in your drone CI for mysql5.7 and mysql8.0, I would be happy to add them for some MariaDB versions if you think that this would make sense and complete this PR...
> 
> I think those would be great to add!

I have prepared https://github.com/grafana/grafana/pull/72013 with https://github.com/grafana/grafana/pull/72013/commits/0950d65e53292acacc03afef81b77f8f37d29996 that permits one to create a local MariaDB docker stack and run integration tests against MariaDB. I am probably not allowed to modify the drone CI (if I understand correctly https://github.com/grafana/grafana/blob/main/contribute/drone-pipeline.md). So I will probably need some guidance there.

mdvictor commented on docs/sources/datasources/mysql/_index.md:
```diff
@@ -2,26 +2,27 @@
-menuTitle: MySQL
-title: MySQL data source
+menuTitle: MySQL/MariaDB
+title: MySQL/MariaDB data source
```
I agree that we could probably specify support for MariaDB more clearly in the docs (although if, in the future, it moves away from MySQL compatibility our DS might not support it anymore), but I don't think we should append `MariaDB` to all occurrences. I'd see this as more of a sentence somewhere in the docs saying that other MySQL replacements like MariaDB also work.

I think that support for MariaDB is mainly the result of it being a compatible drop-in replacement for MySQL. I haven't tested, but I would assume the same could be said about other similar products like [Percona Server](https://www.percona.com/software/mysql-database/percona-server). The main thing I'm trying to point out here is that there may be other MySQL derivates that the DS can connect to, but we cannot change its name in the docs to cater to all.

baldm0mma replied:
This is a good point, and one I didn't fully consider. We DO certainly support MariaDB, but only **_incidentally_**, due to the fact that it maintains compatibility with its MySQL's API and commands. However, if that ever changes, we currently DO NOT have on official MariaDB plugin. @fauust I think @mdvictor is correct in believing that simply appending MariaDB to this section of our docs is misleading, and isn't the correct approach. What do you think of instead mentioning MariaDB's current compatibility here (/docs/sources/datasources/mysql/_index.md, line 24) and here (/public/app/plugins/datasource/mysql/README.md, line 3)?

fauust committed:
Make it clear that MariaDB is also supported as datasource

MariaDB is now the default relational DB on lots of Linux distribution,
it's important to make it clear that it is perfectly supported.

fauust committed:
Fix 404 and make MySQL docs URL more robust

And lint with prettier.

baldm0mma commented:
Thanks @fauust ! I'm going to share this with our technical writers so they can also look it over and approve. :)

baldm0mma reviewed (approved):
LGTM, but let's wait for one of our technical writers to also approve, then you can merge! :)

lwandz13 commented on docs/sources/datasources/mysql/_index.md:
```diff
@@ -185,7 +185,7 @@ The response from MySQL can be formatted as either a table or as a time series.
 
 {{% admonition type="note" %}}
-If your table or database name contains a reserved word or a [not permitted character](https://dev.mysql.com/doc/refman/8.0/en/identifiers.html) the editor will put quotes around them. For example a table name like `table-name` will be quoted with backticks `` `table-name` ``.
+If your table or database name contains a reserved word or a [not permitted character](https://dev.mysql.com/doc/en/identifiers.html) the editor will put quotes around them. For example a table name like `table-name` will be quoted with backticks `` `table-name` ``.
```
```suggestion
If your table or database name contains a reserved word or a [prohibited character](https://dev.mysql.com/doc/en/identifiers.html) the editor will put quotes around them. For example a table name like `table-name` will be quoted with backticks `` `table-name` ``.
```

lwandz13 replied:
```suggestion
If your table or database name contains a reserved word or a [prohibited character](https://dev.mysql.com/doc/en/identifiers.html) the editor will put quotes around the name. For example, the name `table-name` will be quoted with backticks  - `` `table-name` ``.
```

lwandz13 commented on docs/sources/datasources/mysql/_index.md:
```diff
@@ -40,20 +40,20 @@ Administrators can also [configure the data source via YAML](#provision-the-data
+| **Session Timezone**          | Specify the time zone used in the database session, such as `Europe/Berlin` or `+02:00`. This is necessary, if the timezone of the database (or the host of the database) is set to something other than UTC. Set the value used in the session with `SET time_zone='...'`. If you leave this field empty, then the time zone is not updated. For more information, refer to  ...
+| **Max open**                  | The maximum number of open connections to the database, default `100` (Grafana v5.4+).                                                                                                                                                                                                                                                                                        ...
+| **Max idle**                  | The maximum number of connections in the idle connection pool, default `100` (Grafana v5.4+).                                                                                                                                                                                                                                                                                 ...
+| **Auto (max idle)**           | If set will set the maximum number of idle connections to the number of maximum open connections (Grafana v9.5.1+). Default is `true`.                                                                                                                                                                                                                                        ...
```
```suggestion
| **Auto (max idle)**           | Toggle to set the maximum number of idle connections to the number of maximum open connections (available in Grafana v9.5.1+). Default is `true`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
```

lwandz13 commented on docs/sources/datasources/mysql/_index.md:
```diff
@@ -40,20 +40,20 @@ Administrators can also [configure the data source via YAML](#provision-the-data
+| **Max open**                  | The maximum number of open connections to the database, default `100` (Grafana v5.4+).                                                                                                                                                                                                                                                                                        ...
+| **Max idle**                  | The maximum number of connections in the idle connection pool, default `100` (Grafana v5.4+).                                                                                                                                                                                                                                                                                 ...
+| **Auto (max idle)**           | If set will set the maximum number of idle connections to the number of maximum open connections (Grafana v9.5.1+). Default is `true`.                                                                                                                                                                                                                                        ...
+| **Allow cleartext passwords** | Allows using the [cleartext client side plugin](https://dev.mysql.com/doc/en/cleartext-pluggable-authentication.html) if required by an account, such as one defined with the [PAM authentication plugin](https://dev.mysql.com/doc/en/pam-pluggable-authentication.html). <br />**Sending passwords in clear text may be a security problem in some configurations**. To avo ...
```
```suggestion
| **Allow cleartext passwords** | Allows the use of the [cleartext client side plugin](https://dev.mysql.com/doc/en/cleartext-pluggable-authentication.html) as required by a specific type of account, such as one defined with the [PAM authentication plugin](https://dev.mysql.com/doc/en/pam-pluggable-authentication.html). <br />**Sending passwords in clear text may be a security problem in some configurations**. To avoid password issues, it is recommended that clients connect to MySQL server using a method that protects the password. Possibilities include [TLS / SSL](https://github.com/go-sql-driver/mysql#tls), IPsec, or a private network. Default is `false`. |
```

lwandz13 commented on docs/sources/datasources/mysql/_index.md:
```diff
@@ -40,20 +40,20 @@ Administrators can also [configure the data source via YAML](#provision-the-data
+| **Max idle**                  | The maximum number of connections in the idle connection pool, default `100` (Grafana v5.4+).                                                                                                                                                                                                                                                                                 ...
+| **Auto (max idle)**           | If set will set the maximum number of idle connections to the number of maximum open connections (Grafana v9.5.1+). Default is `true`.                                                                                                                                                                                                                                        ...
+| **Allow cleartext passwords** | Allows using the [cleartext client side plugin](https://dev.mysql.com/doc/en/cleartext-pluggable-authentication.html) if required by an account, such as one defined with the [PAM authentication plugin](https://dev.mysql.com/doc/en/pam-pluggable-authentication.html). <br />**Sending passwords in clear text may be a security problem in some configurations**. To avo ...
+| **Max lifetime**              | The maximum amount of time in seconds a connection may be reused, default `14400`/4 hours. This should always be lower than configured [wait_timeout](https://dev.mysql.com/doc/en/server-system-variables.html#sysvar_wait_timeout) in MySQL (Grafana v5.4+).                                                                                                                ...
```
```suggestion
| **Max lifetime**              | The maximum amount of time in seconds a connection may be reused. This should always be lower than configured [wait_timeout](https://dev.mysql.com/doc/en/server-system-variables.html#sysvar_wait_timeout) in MySQL (Grafana v5.4+).   The default is `14400` or 4 hours.                                                                                                                                                                                                                                                                                                                                                                                                            |
```

lwandz13 commented on docs/sources/datasources/mysql/_index.md:
```diff
@@ -40,20 +40,20 @@ Administrators can also [configure the data source via YAML](#provision-the-data
+| **Database**                  | Name of your MySQL database.                                                                                                                                                                                                                                                                                                                                                  ...
+| **User**                      | Database user's login/username                                                                                                                                                                                                                                                                                                                                                ...
+| **Password**                  | Database user's password                                                                                                                                                                                                                                                                                                                                                      ...
+| **Session Timezone**          | Specify the time zone used in the database session, such as `Europe/Berlin` or `+02:00`. This is necessary, if the timezone of the database (or the host of the database) is set to something other than UTC. Set the value used in the session with `SET time_zone='...'`. If you leave this field empty, then the time zone is not updated. For more information, refer to  ...
```
```suggestion
| **Session Timezone**          | Specifies the time zone used in the database session, such as `Europe/Berlin` or `+02:00`. Required if the timezone of the database (or the host of the database) is set to something other than UTC. Set the value used in the session with `SET time_zone='...'`. If you leave this field empty, then the time zone is not updated. For more information, refer to the [MySQL documentation](https://dev.mysql.com/doc/en/time-zone-support.html).                                                                                                                                                                                                                   |
```

lwandz13 reviewed (commented):
a few suggestions

lwandz13 commented:
I added the 10.0 and 10.1 backports....please remove if that's not necessary

baldm0mma commented:
Thanks @lwandz13 ! @fauust feel free to merge once the changes are made.

fauust committed:
Review from lwandz13

fauust commented:
done

baldm0mma commented:
Great! Thanks @fauust ; I'll merge as soon as I figure out why the `Check for patch conflicts / trigger_downstream_patch_check / test-patches (pull_request)` check is failing.
